src/worker.ts

```diff
@@ -1,186 +1,13 @@
-		if (err instanceof RateLimitingError) {
-			// delaying until the rate limit is reset (plus a buffer of a couple seconds)
-			const delay = err.rateLimitReset * 1000 + 10000 - new Date().getTime();
-
-			if (delay <= 0) {
-				logger.warn({ job }, "Rate limiting detected but couldn't calculate delay, delaying exponentially");
-				job.opts.backoff = {
-					type: "exponential",
-					delay: 10 * 60 * 1000
-				};
-			} else {
-				logger.warn({ job }, `Rate limiting detected, delaying job by ${delay} ms`);
-				job.opts.backoff = {
-					type: "fixed",
-					delay: delay
-				};
-			}
-		}
-		throw err;
-	}
-};
-
-const sendQueueMetrics = async () => {
-	if (await booleanFlag(BooleanFlags.EXPOSE_QUEUE_METRICS, false)) {
-
-		for (const [queueName, queue] of Object.entries(queues)) {
-			logger.info("fetching queue metrics");
-
-			const jobCounts = await queue.getJobCounts();
-
-			logger.info({ queue: queueName, queueMetrics: jobCounts }, "publishing queue metrics");
-
-			const tags = { queue: queueName };
-			statsd.gauge(redisQueueMetrics.active, jobCounts.active, tags);
-			statsd.gauge(redisQueueMetrics.completed, jobCounts.completed, tags);
-			statsd.gauge(redisQueueMetrics.delayed, jobCounts.delayed, tags);
-			statsd.gauge(redisQueueMetrics.failed, jobCounts.failed, tags);
-			statsd.gauge(redisQueueMetrics.waiting, jobCounts.waiting, tags);
-		}
-	}
-};
-
-const commonMiddleware = (jobHandler, loggerName: string) => logMiddleware(sentryMiddleware(setDelayOnRateLimiting(jobHandler)), loggerName);
-
-let running = false;
-let timer: Timeout;
-
-// Start function for Node cluster worker
-async function start() {
-	if (running) {
-		logger.debug("Worker instance already running, skipping.");
-		return;
-	}
-
-	logger.info("Micros Lifecycle: Starting queue processing");
-	// exposing queue metrics at a regular interval
-	timer = setInterval(sendQueueMetrics, 60000);
-
-	// Start processing queues
-	queues.discovery.process(5, commonMiddleware(discovery(app, queues), DISCOVERY_LOGGER_NAME));
-	queues.installation.process(
-		Number(CONCURRENT_WORKERS),
-		commonMiddleware(processInstallation(app, queues), INSTALLATION_LOGGER_NAME)
-	);
-	queues.push.process(
-		Number(CONCURRENT_WORKERS),
-		commonMiddleware(processPushJob(app), PUSH_LOGGER_NAME)
-	);
-	queues.metrics.process(1, commonMiddleware(metricsJob, METRICS_LOGGER_NAME));
-
-	sqsQueues.start()
-
-	running = true;
-}
-
-async function stop() {
-	if (!running) {
-		logger.debug("Worker instance not running, skipping.");
-		return;
-	}
-	logger.info("Micros Lifecycle: Stopping queue processing");
-	// TODO: change this to `probot.close()` once we update probot to latest version
-	probot.httpServer?.close();
-
-	// Stop sending metrics for queues
-	clearInterval(timer);
-
-	// Close all queues
-	await Promise.all([
-		queues.discovery.close(),
-		queues.installation.close(),
-		queues.push.close(),
-		queues.metrics.close()
-	]);
-
-	sqsQueues.stop();
-
-	running = false;
-}
```

Comment:
Moved this functions to `src/worker/startup.ts` so we can use them without triggering the worker startup logic